### PR TITLE
reset control mode for pybullet robots

### LINKED ISSUE #839: pybullet robots: make flag to toggle using setmotors vs. directly reseting the joint states
the latter will be useful for debugging (and may ultimately be all we need...)

### PR DESCRIPTION
**Pull request by tomsilver**
closes #839

### CODE REVIEW

**Comment by ronuchit** [comment]
after rebasing off #848 the correct code is:
```
            world_to_base_link = p.getLinkState(
                self._pybullet_robot.robot_id,
                self._pybullet_robot.end_effector_id,
                physicsClientId=self._physics_client_id)[:2]
            base_link_to_held_obj = p.invertTransform(*self._held_obj_to_base_link)
            world_to_held_obj = p.multiplyTransforms(
                world_to_base_link[0], world_to_base_link[1],
                base_link_to_held_obj[0], base_link_to_held_obj[1])
```

File: src/envs/pybullet_robots.py [hunk 1]
```diff
@@ -346,11 +346,23 @@ def set_motors(self, action_arr: Array) -> None:
...
-                                    physicsClientId=self._physics_client_id)
+        if CFG.pybullet_control_mode == "position":
+            p.setJointMotorControlArray(
+                bodyUniqueId=self._fetch_id,
+                jointIndices=self._arm_joints,
+                controlMode=p.POSITION_CONTROL,
+                targetPositions=action_arr,
+                physicsClientId=self._physics_client_id)
+        elif CFG.pybullet_control_mode == "reset":
+            for joint_id, joint_val in zip(self._arm_joints, action_arr):
```
Comment (tomsilver) [inline src/envs/pybullet_robots.py:357]: we'll see who merges first...

**Review by tomsilver — COMMENTED** [review]
lgtm. did you try out oracle in pybullet blocks and cover?

**Review by ronuchit — APPROVED** [review]

**Comment by ronuchit** [comment]
supercloud results after fix in #850:
```
Git commit hashes seen in results/:
4109445d6bf8655b90702d6e0fdf9f83c34a21f8


AGGREGATED DATA OVER SEEDS:
                                        NUM_TRAIN_TASKS    NUM_SOLVED AVG_NUM_PREDS AVG_TEST_TIME  AVG_NODES_CREATED      LEARNING_TIME  NUM_SEEDS
EXPERIMENT_ID
pybullet_blocks_invent_allexclude_reset    50.00 (0.00)  49.60 (0.66)   6.50 (1.12)   0.44 (0.13)  4331.86 (2137.50)   2323.63 (273.72)         10
pybullet_blocks_invent_noexclude_reset     50.00 (0.00)  49.60 (0.66)   6.10 (1.58)   0.44 (0.17)  4024.07 (1879.06)  1339.77 (1450.43)         10
pybullet_blocks_nsrt_learning_reset        50.00 (0.00)  49.90 (0.30)   5.00 (0.00)   0.37 (0.11)  4195.92 (2119.92)       17.05 (1.32)         10
pybullet_blocks_oracle_reset                0.00 (0.00)  50.00 (0.00)   5.00 (0.00)   0.37 (0.11)  4259.69 (2224.80)        0.00 (0.00)         10
pybullet_cover_invent_allexclude_reset     50.00 (0.00)  49.20 (0.60)   5.90 (0.54)   0.04 (0.00)        5.01 (0.24)     244.95 (27.19)         10
pybullet_cover_invent_noexclude_reset      50.00 (0.00)  49.20 (0.60)   7.00 (0.45)   0.04 (0.01)        5.01 (0.24)     156.27 (22.37)         10
pybullet_cover_nsrt_learning_reset         50.00 (0.00)  49.30 (0.46)   5.00 (0.00)   0.09 (0.02)        6.67 (0.34)       21.39 (1.45)         10
pybullet_cover_oracle_reset                 0.00 (0.00)  49.70 (0.46)   5.00 (0.00)   0.07 (0.01)        6.66 (0.32)        0.00 (0.00)         10
```

**Pull request merged**